In [1]:
import pandas as pd
import numpy as np


In [2]:
df=pd.read_csv("preprocesamiento base de datos.csv")

De acuerdo a las 4 bases de datos entregadas por Soy Tutor, se hace una unión de éstas con las columnas que consideramos que son clave para poder analizar una tendencia sobre el performance del tutor.

El diccionario de variables quedaría de la siguiente manera:

**ID_Tutor**: Se refiere al id asociado a cada uno de los tutores

**Correo_tutor**: Es el correo de los tutores que estan desplegados en la lista

**carrera_tutor**: Es la carrera que actualmente cursa el tutor

**tipo_tutor**: Se refiere al tipo de tutor, para este caso existen 3 tipos de tutores distintos :

1 = Tutor: Servicio Becario

2 = Tutor: Servicio Social

6= Tutor: Tec Milenio

**Status_tutoría**:  Es el estatus de la tutoria, existen 11 estatus diferentes

-10 = Ninguno (tutor y alumno) se conectó a la sesión de tutoring.

-9 = Tutor NO se conectó a la sesión y Alumno SÍ se conectó.

-8 = Alumno NO se conectó a la sesión y Tutor SÍ se conectó.

-7 = Alumno en espera de que Tutor se conecte a la sesión de tutoring.

-6 = Tutor en espera de que Alumno se conecte a la sesión de tutoring.

-3 = Sesión de tutoring cancelada por Tutor.

-1 = Sesión pendiente de aceptar/rechazar por parte del Tutor. (No Aplica)

0 = Sesión de tutoring aceptada. (En automático)

2 = Sesión cancelada.

4 = Sesión de tutoring finalizada exitosamente

5 = Sesión finalizada por el Tutor. (antes de tiempo)

**IDAppointment**: Es el Id asociado a cada una de las retroalimentaciones dadas por los tutores o alumnos

**categoria_principal**: Es la primera categoria de las tutorias que puede impartir el tutor

**subcategoria_principal**: Es la primera subcategoria de las tutorias que puede impartir el tutor

**categoria_secundaria**: Es la segunda categoria de las tutorias que puede impartir el tutor

**subcatgoria_secundaria**: Es la segunda subcategoria de las tutorias que puede impartir el tutor

**Desempeño_Tutor**: es la calificación obtenida por el tutor en su evaluación bajo el tipo de pregunta Estrellas = Rango es del 1-5 donde 1 es la califación más baja y 5 la más alta.

**Meta_Tutoría**: es la calificación donde el alumno percibe si la tutoría alcanzó sus objetivos deseados bajo el tipo de pregunta Radio = Rango es del 1-5 (Muy en Desacuerdo, En Desacuerdo, Neutral, De Acuerdo, Muy en Deacuerdo)

In [3]:
df.head()

ID_Tutor      correo_tutor  \
0     14566  a01178149@tec.mx   
1     19866  a01750754@tec.mx   
2     15823  a01385118@tec.mx   
3     14074  a01658389@tec.mx   
4     19886  A01745719@tec.mx   

                                       carrera_tutor tipo_tutor  \
0  Ingenier’a - Bioingenier’a y Procesos Qu’micos...          1   
1                                               #N/D          1   
2                                               #N/D       #N/D   
3                                               #N/D       #N/D   
4                                               #N/D       #N/D   

   Status_Tutoría  IDAppointment categoria_principal  \
0               4           8361             Química   
1               4           8504         Matemáticas   
2               4           7754                #N/D   
3               4           7107                #N/D   
4               4           8609                #N/D   

                              subcategoria_principal  \
0  Enlaces químicos e interacciones intermolecucl...   
1                        Magnitudes y números reales   
2                                               #N/D   
3                                               #N/D   
4                                               #N/D   

           categoría_secundaria  \
0                   Matemáticas   
1  Preparación a la Universidad   
2                          #N/D   
3                          #N/D   
4                          #N/D   

                             subcategoria_secundaria Desempeño_Tutor  \
0  Cálculo de derivadas mediante reglas, fórmulas...               1   
1                                              ITESM               5   
2                                               #N/D            #N/D   
3                                               #N/D               5   
4                                               #N/D               5   

  Meta_Tutoría  
0         #N/D  
1            5  
2         #N/D  
3            0  
4            5

In [4]:
df.count()


ID_Tutor                   11463
correo_tutor               11463
carrera_tutor              11463
tipo_tutor                 11463
Status_Tutoría             11463
IDAppointment              11463
categoria_principal        11463
subcategoria_principal     11463
categoría_secundaria       11463
subcategoria_secundaria    11463
Desempeño_Tutor            11463
Meta_Tutoría               11463
dtype: int64

In [5]:
df.describe()

ID_Tutor  Status_Tutoría  IDAppointment
count  11463.000000    11463.000000   11463.000000
mean   15161.823345        4.025735    7788.061502
std     2284.171680        0.839991     713.711349
min    10075.000000      -10.000000    6289.000000
25%    13957.000000        4.000000    7259.000000
50%    14584.000000        4.000000    7751.000000
75%    15484.000000        4.000000    8360.000000
max    22382.000000        5.000000    9456.000000

In [6]:
#revisamos si existen datos faltante en alguna columna

df.isnull().any()

ID_Tutor                   False
correo_tutor               False
carrera_tutor              False
tipo_tutor                 False
Status_Tutoría             False
IDAppointment              False
categoria_principal        False
subcategoria_principal     False
categoría_secundaria       False
subcategoria_secundaria    False
Desempeño_Tutor            False
Meta_Tutoría               False
dtype: bool

In [7]:
df.dtypes

ID_Tutor                    int64
correo_tutor               object
carrera_tutor              object
tipo_tutor                 object
Status_Tutoría              int64
IDAppointment               int64
categoria_principal        object
subcategoria_principal     object
categoría_secundaria       object
subcategoria_secundaria    object
Desempeño_Tutor            object
Meta_Tutoría               object
dtype: object

In [8]:
#se reemplaza los valores "#N/D" por un valor nulo para considerarlo como valor faltante

df.replace("#N/D",np.nan, inplace=True)

In [18]:
#eliminamos las filas que contengan algún valor faltante.

df_final=df.dropna()

In [19]:
#validamos la cantidad de valores por cada una de las columnas una vez eliminado los valores faltantes

df_final.count()

ID_Tutor                   1128
correo_tutor               1128
carrera_tutor              1128
tipo_tutor                 1128
Status_Tutoría             1128
IDAppointment              1128
categoria_principal        1128
subcategoria_principal     1128
categoría_secundaria       1128
subcategoria_secundaria    1128
Desempeño_Tutor            1128
Meta_Tutoría               1128
dtype: int64

Una vez eliminado todos los valores faltantes y que la base de datos es reducida pordemos empezar a analizar algunas tendencias sobre los datos a procesar.

Pero antes es importante mencionar que la reducción de la base de datos fue significativa ya que de las poco más de 11 mil filas, nos quedamos con soamente 1128 filas a analizar

La gran mayoría de los tutores que están registrados en la plataforma, están trabajando como parte de su servicio de becario.


In [20]:
df_final["tipo_tutor"].value_counts()

1    1120
2       8
Name: tipo_tutor, dtype: int64

predominan las tutorías que fueron terminandas con éxito en tiempo y forma, aunque el status con más menciones que le sigue es el que las tutorías se dieron pero acabaron antes de tiempos

In [21]:
df_final["Status_Tutoría"].value_counts()

 4     1054
 5       65
-10       9
Name: Status_Tutoría, dtype: int64

La especialidad principal de los tutores es de Matemáticas

In [22]:
df_final["categoria_principal"].value_counts()

Matemáticas                     811
Física                          111
Química                          94
Programación                     64
Preparación a la Universidad     35
Soporte Técnico                  13
Name: categoria_principal, dtype: int64

En general, los alumnos están satisfechos con el desempeño de sus tutores.

In [23]:
df_final["Desempeño_Tutor"].value_counts()

5    1055
4      63
2       5
3       5
Name: Desempeño_Tutor, dtype: int64

En general, los alumnos perciben que sus objetivos académicos por los que asisten a la tutoría son cumplidos.

In [24]:
df_final["Meta_Tutoría"].value_counts()

5    679
4    312
3     67
0     55
2     10
1      5
Name: Meta_Tutoría, dtype: int64

In [25]:
df_final.head()

ID_Tutor      correo_tutor  \
14     12021  a01027074@tec.mx   
15     14869  A00835528@tec.mx   
18     14213  A01285253@tec.mx   
38     12024  a00835084@tec.mx   
40     12281  A01275608@tec.mx   

                                        carrera_tutor tipo_tutor  \
14  Derecho, Econom’a y Relaciones Internacionales...          1   
15  Ingenier’a - Innovaci—n y Transformaci—n (aven...          1   
18                        Licenciado en Mercadotecnia          1   
38  Ingenier’a - Innovaci—n y Transformaci—n (aven...          1   
40  Derecho, Econom’a y Relaciones Internacionales...          1   

    Status_Tutoría  IDAppointment categoria_principal  \
14               4           6825         Matemáticas   
15               4           8048             Física    
18               4           7332         Matemáticas   
38               4           7724         Matemáticas   
40               4           6525         Matemáticas   

                               subcategoria_principal  \
14  Cálculo de derivadas mediante reglas, fórmulas...   
15                           Leyes de los movimientos   
18                                Ecuaciones lineales   
38                                         Integrales   
40   Ángulos, triángulos y relaciones trigonométricas   

            categoría_secundaria subcategoria_secundaria Desempeño_Tutor  \
14                   Matemáticas              Integrales               5   
15             Ciencias Sociales            Programación               5   
18  Preparación a la Universidad                   ITESM               5   
38                   Matemáticas                 Calculo               5   
40                   Matemáticas     Ecuaciones lineales               5   

   Meta_Tutoría  
14            5  
15            5  
18            0  
38            0  
40            5

In [29]:
df_final.dtypes

ID_Tutor                    int64
correo_tutor               object
carrera_tutor              object
tipo_tutor                 object
Status_Tutoría              int64
IDAppointment               int64
categoria_principal        object
subcategoria_principal     object
categoría_secundaria       object
subcategoria_secundaria    object
Desempeño_Tutor            object
Meta_Tutoría               object
dtype: object

In [33]:
df_final["tipo_tutor"]=df_final["tipo_tutor"].astype("int64")

<ipython-input-33-326b291f4a89>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_final["tipo_tutor"]=df_final["tipo_tutor"].astype("int64")


In [34]:
df_final["Desempeño_Tutor"]=df_final["Desempeño_Tutor"].astype("int64")

<ipython-input-34-2fe9af29ceb9>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_final["Desempeño_Tutor"]=df_final["Desempeño_Tutor"].astype("int64")


In [35]:
df_final["Meta_Tutoría"]=df_final["Meta_Tutoría"].astype("int64")

<ipython-input-35-1c724e1128cd>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_final["Meta_Tutoría"]=df_final["Meta_Tutoría"].astype("int64")


In [37]:
df_final.dtypes

ID_Tutor                    int64
correo_tutor               object
carrera_tutor              object
tipo_tutor                  int64
Status_Tutoría              int64
IDAppointment               int64
categoria_principal        object
subcategoria_principal     object
categoría_secundaria       object
subcategoria_secundaria    object
Desempeño_Tutor             int64
Meta_Tutoría                int64
dtype: object

In [38]:
df_final.corr()

<ipython-input-38-dcfd88b6da59>:1: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  df_final.corr()


ID_Tutor  tipo_tutor  Status_Tutoría  IDAppointment  \
ID_Tutor         1.000000   -0.126224        0.025470       0.622149   
tipo_tutor      -0.126224    1.000000       -0.428214      -0.140004   
Status_Tutoría   0.025470   -0.428214        1.000000       0.089653   
IDAppointment    0.622149   -0.140004        0.089653       1.000000   
Desempeño_Tutor  0.011415    0.020120       -0.010119      -0.030069   
Meta_Tutoría    -0.169424    0.048534       -0.068319      -0.209379   

                 Desempeño_Tutor  Meta_Tutoría  
ID_Tutor                0.011415     -0.169424  
tipo_tutor              0.020120      0.048534  
Status_Tutoría         -0.010119     -0.068319  
IDAppointment          -0.030069     -0.209379  
Desempeño_Tutor         1.000000      0.222403  
Meta_Tutoría            0.222403      1.000000